### DISCUSSION TITLE
handle cracked/offline clients

### DISCUSSION THREAD

**Discussion by bananasmoothii** [Minestom/Minestom#1224] [Help]
I just found out `MojangAuth.init()` for premium clients, but this makes it impossible for offline clients to join. How can I show them a proper error message ? And how could I let them still log in ?

**Comment by ZakShearman** [comment]
If you want offline users to join, you can't enable mojang auth. Enabling mojang auth is the equivalent to online-mode=true on Vanilla based servers.

**Reply by bananasmoothii** [reply]
I just want the uuid to match the premium uuid and the skin/capes to be synchronized for premium clients. Is it possible otherwise ?

**Reply by ryleu** [reply]
@bananasmoothii you’re asking for a way to allow cracked players to join while still using mojang services for skins? if so, that’s against the minecraft ToS, and you won’t find a way to do it here.

**Reply by bananasmoothii** [reply]
Oops, ok...

**Reply by bananasmoothii** [reply]
So how do I display a custom error message ?
